Assets/SEE/Controls/KeyBindings.cs

```diff
@@ -1,438 +1,268 @@
             {
-                Debug.LogError($"Cannot register key {keyCode} for [{scope}] {helpMessage}\n");
-                Debug.LogError($"Key {keyCode} already bound to {bindings[keyCode]}\n");
+                keyCode = descriptor.KeyCode;
+                return true;
             }
             else
             {
-                bindings[keyCode] = $"[{scope}] {helpMessage}";
+                keyCode = KeyCode.None;
+                return false;
             }
-            return keyCode;
         }
+
         /// <summary>
-        /// Returns the scope of given key-binding description <paramref name="value"/>.
+        /// True if given <paramref name="key"/> can be re-assigned by a user.
         /// </summary>
-        /// <param name="value">the key-binding description from which to extract the scope</param>
-        /// <returns>the scope</returns>
-        private static string GetScope(string value)
+        /// <param name="key">the key code to be checked</param>
+        /// <returns>True if given <paramref name="key"/> can be re-assigned by a user.</returns>
+        internal static bool AssignableKeyCode(KeyCode key)
         {
-            // Extract the scope part from the string value.
-            int startIndex = value.IndexOf("[") + 1;
-            int endIndex = value.IndexOf("]");
-            return value[startIndex..endIndex];
+            // We are using the mouse buttons 0, 1, and 2 for other purposes. We do not want
+            // them to be re-assigned.
+            return key != KeyCode.Mouse0 && key != KeyCode.Mouse1 && key != KeyCode.Mouse2;
         }
+
         /// <summary>
-        /// Prints the current key bindings to the debugging console along with their
-        /// help message.
+        /// Registers the given <paramref name="keyCode"/> for the given <paramref name="scope"/>
+        /// and the <paramref name="description"/>. If a <paramref name="keyCode"/> is already registered,
+        /// an error message will be emitted.
         /// </summary>
-        internal static void PrintBindings()
+        /// <param name="keyCode">the key code to be registered</param>
+        /// <param name="name">the name of the action triggered by the <paramref name="keyCode"/>;
+        /// <param name="scope">the scope of the key code</param>
+        /// <param name="description">the help message for the key code</param>
+        /// this name will be presented to our users</param>
+        private static void Register(KeyAction keyAction, KeyCode keyCode, string name, KeyActionCategory scope, string description)
         {
-            System.Text.StringBuilder sb = new("Key Bindings:\n");
-            foreach (var binding in bindings)
+            if (keyBindings.TryGetValue(keyAction, out KeyActionDescriptor descriptor))
+            {
+                Debug.LogError($"Key action {keyAction} is already bound ot {descriptor.Name}.\n");
```

Comment:
Are these intended to be displayed to the user (e.g., for reassigment)? If so, we should probably use `ShowNotification` here instead.

Reply:
I fixed it.